web/src/people.ts

```diff
@@ -1934,6 +1947,86 @@ export function populate_valid_user_ids(params: StateData["user_groups"]): void
     }
 }
 
+function get_unique_key_for_user_fetch(user_ids: Set<number>): string {
+    const fetch_unique_key = [...user_ids].sort().join(",");
+    return fetch_unique_key;
+}
+
+function start_fetch_for_requested_users(): void {
+    const user_ids_to_fetch = fetch_users_storage.pending_user_ids;
+    fetch_users_storage.pending_user_ids = new Set();
+    fetch_users_storage.in_transit_user_ids =
+        fetch_users_storage.in_transit_user_ids.union(user_ids_to_fetch);
+    const fetch_unique_key = get_unique_key_for_user_fetch(user_ids_to_fetch);
+    fetch_users_storage.promise_resolvers_for_in_transit.set(
+        fetch_unique_key,
+        fetch_users_storage.promise_resolvers_for_pending,
+    );
+    fetch_users_storage.promise_resolvers_for_pending = [];
+
+    void fetch_users(user_ids_to_fetch).then((fetched_users) => {
+        for (const user of fetched_users) {
+            if (user.is_active) {
+                add_active_user(user);
+            } else {
+                non_active_user_dict.set(user.user_id, user);
+                _add_user(user);
+            }
+        }
+
+        const promise_resolvers =
+            fetch_users_storage.promise_resolvers_for_in_transit.get(fetch_unique_key)!;
+        for (const resolve of promise_resolvers) {
+            resolve();
+        }
```

Comment:
This seems wrong. We don’t know that all of the in-transit users have been fetched at this point; we only know that some of them have.

Reply:
I made some changes here. We are only resolving promise for the users we fetched.